src/ECS/System/RenderSystem.cpp

```diff
@@ -0,0 +1,35 @@
+#include "ECS/System/RenderSystem.h"
+
+namespace Sourcehold {
+    namespace ECS {
+        namespace System {
+            RenderSystem::RenderSystem() {
+                ghc::filesystem::path dataRoot = GetDirectory();
+                lordFile = GetGm1(dataRoot.string() + "gm/body_lord.gm1");
+                deerFile = GetGm1(dataRoot.string() + "gm/body_deer.gm1");
+                appleTreeFile = GetGm1(dataRoot.string() + "gm/tree_apple.gm1");
+                birchTreeFile = GetGm1(dataRoot.string() + "gm/tree_birch.gm1");
+                chestnutTreeFile = GetGm1(dataRoot.string() + "gm/Tree_Chestnut.gm1");
+                oakTreeFile = GetGm1(dataRoot.string() + "gm/tree_oak.gm1");
+                pineTreeFile = GetGm1(dataRoot.string() + "gm/tree_pine.gm1");
+                shrub1TreeFile = GetGm1(dataRoot.string() + "gm/tree_shrub1.gm1");
+                shrub2TreeFile = GetGm1(dataRoot.string() + "gm/tree_shrub2.gm1");
+            }
+
+            void RenderSystem::each(EachEntityType entity) {
```

Comment:
Tbh - when we are in ECS context we aren't "rendering", we are "performing operations on data set".
I can take that out to something else, but I wanted to use BasicSystem where it is not mandatory to define something giga-different.

Reply:
Tbh - when we are in ECS context we aren't "rendering", we are "performing operations on/with/basingon data set".
I can take that out to something else, but I wanted to use BasicSystem where it is not mandatory to define something giga-different.

Reply:
Fair enough.
In C++ functions are usally defined in this format (atleast that's how I was taught):
```C++
// In general, name functions so
// reading from left to right makes sense
void applySomethingOn(data)

// both version are fine. choose the one which is nicer to read
SomeType giveMeWithThisConfig(settings)
auto giveMeWithThisConfig(settings) -> SomeType

// sometimes you need this
void doSomeStuff( data, settings)
// or just
void doStuff()
```
Define classes freely, but use polymorphism only when absolutely necessary.
I think in your case it's fine.
Otherwise, prefer templates:
```C++
template <typename Config>
someGenericFunction();

template <typename Config>
class SomeGenericClass{};
```

Sorry, if I'm a bit picky on this matter

Reply:
I struggled about API naming convention for System for a bit, but I think the best bet will be treating it like "clock".
So... general update is called `tick` and it is called on register, there is `prepareTick` state and `tickEntity`.